server/src/test/java/GroceryFamily/GroceryMom/api/ProductAPITest.java

```diff
@@ -0,0 +1,150 @@
+package GroceryFamily.GroceryMom.api;
+
+import GroceryFamily.GroceryElders.domain.Page;
+import GroceryFamily.GroceryElders.domain.Price;
+import GroceryFamily.GroceryElders.domain.Product;
+import GroceryFamily.GroceryMom.service.ProductService;
+import GroceryFamily.GroceryMom.service.exception.ProductNotFoundException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.client.HttpServerErrorException;
+
+
+import java.math.BigDecimal;
+import java.util.*;
+
+import static org.hamcrest.Matchers.is;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+class ProductAPITest {
+
+    private MockMvc mockMvc;
```

Comment:
This is not needed here.

Reply:
Why don't we need MockMvc here? I think that if we need to check HTTP requests, then we should use MockMvc

Reply:
These are the API unit tests, so you need to fairly create `ProductAPI` as follows:

```java
void setUp() {
    ...
    productAPI = new ProductAPI(service);
    ...
}
```

And then you simply invoke its methods: 

```java
@Test
void test() {
    ...
    when(service.list(...)).thenReturn(...);
    ...
    var products = productAPI.list(...);
    // assert products, validate more pages etc.
    ...
}
```